### PR TITLE
Adding docs readme and getting started guide

### PR DESCRIPTION
**Pull request by pokearu**
## Description
Adding documentation to elaborate on rufio API and a getting started guide.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# Rufio
+## Whos is Rufio?
```
Comment (jacobweinstock) [inline docs/README.md:2]: Whos -> Who?

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    host: 0.0.0.0
+    port: 623
+    authSecretRef:
+      name: bm-auth
+      namespace: sample
+    insecureTLS: false
+```
+The `connection` object contains the required fields for establising a BMC connection. Fields `host`, `port` represent the BMC IP for the physical machine and `insecureTLS` instructs weather to use insecure TLS connectivity for performing BMC API calls. Field `authSecretRef` is a `SecretReference` which points to a kubernetes secret that contains the username/password for authenticating BMC API calls.
+### Machine controller
+When a Machine object is created on the cluster, the machine controller is responsible for updating the current state of the physical machine. It performs redfish/ipmitool API calls to the BMC on the physical machine and updates the `status` of the Machine object.
```
Comment (jacobweinstock) [inline docs/README.md:25]: bmclib abstracts away the protocols used to interact with a BMC. might think about making this statement (`redfish/ipmitool API calls`) more generic.

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+### Job API
+The Job type is used to define a set of one-off operations/actions to be performed on a physical machine. These actions are performed utalizing refish/ipmitool based BMC API calls.
```
Comment (jacobweinstock) [inline docs/README.md:27]: spelling. `utalizing`

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+```
+We can now interact with Rufio using its API. Before creating a Mahine object, we need to create a k8s Secret that contains the BMC user credentials for the physical machine. An example Secret,
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: bm-auth
+  namespace: sample
+data:
+  username: YWRtaW4=
```
Comment (jacobweinstock) [inline docs/README.md:94]: i know this is just an example, but maybe this should be something like `root`, etc?

**Review by jacobweinstock — COMMENTED** [review]
This is great, thanks for putting this together @pokearu! I left a few comments.

File: docs/README.md (hunk 2 above)
Comment (pokearu) [reply docs/README.md:25]: Good point. Will remove that.

File: docs/README.md (hunk 4 above)
Comment (pokearu) [reply docs/README.md:94]: Sure! I just grabbed this off k8s docs.

**Review by jacobweinstock — APPROVED** [review]

**Review by chrisdoherty4 — COMMENTED** [review]
Useful doc. Would it be better off as a section in the root README so people don't need to navigate to this separately.

**Pull request merged**